### ISSUE TITLE
Adding a RaySGD and DGL ( Deep Graph Library) integration example? 

### ISSUE THREAD

**Issue by YeahNew** [ray-project/ray#15552]
We are currently using DGL（https://github.com/dmlc/dgl） to do graph neural network research and see that raySGD has good support for distributed DL training, so we tried to combine DGL's model training with raySGD and succeeded. Under the same training, almost the same accuracy rate was finally obtained. Can you add this case as an application examplet to the project?

**Comment by amogkam** [comment]
Wow this sounds amazing! Do you mind sharing more details on how the integration works? We'd certainly be happy to add it as an example.

**Comment by YeahNew** [comment]
> Wow this sounds amazing! Do you mind sharing more details on how the integration works? We'd certainly be happy to add it as an example.
@amogkam 
OK, Do I need to submit a Pull request? Or show the code directly here?

**Comment by richardliaw** [comment]
@YeahNew could you link the code here? That way we could provide best guidance for how to submit the PR!

**Comment by YeahNew** [comment]
> @YeahNew could you link the code here? That way we could provide best guidance for how to submit the PR!
@richardliaw @amogkam 
yes, the link is here: https://github.com/YeahNew/RayAndDGL.
But it should be noted that simply using DGL (GATwithDGL.py), the script can only run on one GPU.
The case using RaySGD can be run on multiple GPUs.
When multiple GPUs load data, the result of single GPU validation may be slightly different due to data is shuffed.

**Comment by YeahNew** [comment]
> @YeahNew could you link the code here? That way we could provide best guidance for how to submit the PR!

**Comment by richardliaw** [comment]
OK that seems great. Do you want to clean up your code (maybe run a linter like flake8 or black) and link it into the documentation as an example for raysgd?

Also, did you have any feedback for RaySGD?

**Comment by YeahNew** [comment]
> OK that seems great. Do you want to clean up your code (maybe run a linter like flake8 or black) and link it into the documentation as an example for raysgd?
> 
> Also, did you have any feedback for RaySGD?
@richardliaw @amogkam 
Okay, these days I have worked hard to clean up the code and solve the problem of incorrect validation accuracy. Do I need to submit a PR after completing the modification?
Feedback：RaySGD is a great job. It is very convenient to encapsulate the model in the CustomTrainingOperator class, and then create the corresponding actor class to perform distributed parallel computing. But I see that the communication between processes in DL training still uses the backend of the original DL framework itself, instead of using Ray’s own ObjectStore shared memory mechanism, so can RaySGD consider using ObjectStore to implement process communication during DL training? Compare the efficiency between the two. Thanks, this is what I want to know.

**Comment by amogkam** [comment]
Yes please feel free to make a PR and tag us.

**Comment by YeahNew** [comment]
> Yes please feel free to make a PR and tag us.

Can you help me find out the reason for the inconsistency between the final validation results of these two comparison scripts（https://github.com/YeahNew/RayAndDGL.）?
I carefully compared the execution logic of the validate phase and found that there was no difference, but I couldn't find the reason for the huge difference in the results.

**Comment by YeahNew** [comment]
> @YeahNew Your link is broken?

No, I’m on vacation recently, and I started working today.

**Comment by YeahNew** [comment]
> @YeahNew Your link is broken?

hello, This link can be accessed:
https://github.com/YeahNew/RayAndDGL
the code(GATwithRayAndDGLonMultipleGPUs.py) has been modified well, The validation result is almost the same as the result of using only DGL(GATwithDGLonMultipleGPUs.py).
But one thing to note at present is that under the same parameter settings, the training time of each epoch of the former is significantly greater than the training time of each epoch of the latter.
The reason is not clear, it makes me very confused.
The RaySGD and DGL integration case is basically completed, can I submit a PR now?

**Comment by YeahNew** [comment]
@richardliaw @amogkam 
Hello,
I have submitted a PR, please take a look .
https://github.com/ray-project/ray/pull/15718
